docs/guides/best_practices/performance.md

```diff
@@ -1,15 +1,16 @@
 # Performance
 
-## Cache computations with `@functools.cache`
+## Cache computations with `@mo.save.cache`
 
-Use Python's builtin `functools` library to cache expensive computations.
-
-For example,
+You may use `mo.save.cache` to cache expensive computations, in a notebook aware manner.
```

Comment:
> You may use `mo.save.cache` to cache expensive computations, in a notebook aware manner.

What is "notebook aware"? I might rephrase as "You may use the decorator `mo.save.cache` to cache expensive computations. This works similarly to `functools.cache`, but also <invalidates the cache when UI elements, etc change>. 

Additionally, I have a few questions.

1. If my function didn't close over `State` or `UIElement` objects, but instead took their values as arguments:

```
def foo(state_value, ui_element_value):
   ...
```

would using `functools.cache` and `mo.save.cache` yield equivalent results?

2. When the arguments are the UI Element / State objects themselves, what happens?

```
@mo.save.cache
def foo(state_getter, slider):
  ...
```

Does the cache key on the state/slider values, or on their object identities?

3. Does `cache` work when running as a script.

Reply:
Good suggestion, will try to fold it back into. For your Qs:

1. Taking in their values would be exactly the same
2. State is hashable so functools does work for this, but UI elements are not
To your point though
```python
fib(1)
s, _ = state(1)
fib(s) # This is a cache hit as is even though fib(s) would fail by itself
```
This fixable. But not, object identity is not utilized for the hash. This matters when restoring values though

3. Yep!